rsp_jupyter_extensions/handlers/query.py

```diff
@@ -92,24 +106,32 @@ async def _create_query(self, q_value: str, q_type: str) -> str:
 
     async def _create_tap_query(self, q_value: str) -> str:
         # The value should be a URL or a jobref ID
-        this_rsp = os.getenv("EXTERNAL_INSTANCE_URL", "not-an-rsp")
-        if q_value.startswith(this_rsp):
+        # A jobref is always 16 alphanumeric characters.
+        # Therefore: if it contains a slash, it's a URL
+        if q_value.find("/") != -1:
             # This looks like a URL
             url = q_value
             q_id = q_value.split("/")[-1]  # Last component is the jobref ID
+        # If it contains a colon, it's dataset:jobref_id.
+        elif q_value.find(":") != -1:
+            dataset, q_id = q_value.split(":")
+            base_url = get_service_url("tap", dataset)
+            url = f"{base_url}/async/{q_id}"
         else:
-            # It's a raw jobref ID
+            # No colon, so no dataset, so we assume the "/api/tap"
+            # endpoint.
+            this_rsp = os.getenv("EXTERNAL_INSTANCE_URL", "")
             url = f"{this_rsp}/api/tap/async/{q_value}"
             q_id = q_value
         fname = self._root_dir / "notebooks" / "queries" / f"tap_{q_id}.ipynb"
```

Comment:
I think there is a slight chance we get a name collision here if the user tries to load a job that has the same ID across different datasets.

Reply:
I'm going to go back to lsst-rsp and use the last dataset I saw for a given endpoint, on the grounds that people are *probably* looking at "dp1" rather than "dp02".  That rests on some shaky assumptions that newer is later in the discovery document, but...well, it's a heuristic.